rtc_auth_enclave/src/token_store.rs

```diff
@@ -0,0 +1,82 @@
+use std::collections::HashMap;
+use std::io;
+use std::path::Path;
+use std::string::String;
+
+use jwt::EncodedExecutionToken;
+use once_cell::sync::OnceCell;
+use rtc_tenclave::kv_store::fs::{FsStore, SgxFiler};
+use rtc_tenclave::kv_store::KvStore;
+use serde::{Deserialize, Serialize};
+use sgx_tstd::sync::{SgxMutex as Mutex, SgxMutexGuard as MutexGuard};
+use sgx_tstd::untrusted::{fs as untrusted_fs, path as untrusted_path};
+use uuid::Uuid;
+
+use crate::{jwt, uuid_to_string};
+
+#[derive(Serialize, Deserialize)]
+struct ExecutionTokenRecord {
+    exec_module_hash: [u8; 32],
+    dataset_uuid: Uuid,
+    allowed_uses: u32,
+    current_uses: u32,
+}
+
+fn kv_store<'a>(
+) -> MutexGuard<'a, impl KvStore<HashMap<Uuid, ExecutionTokenRecord>, Error = io::Error>> {
+    static TOKEN_FS_STORE: OnceCell<Mutex<FsStore<SgxFiler>>> = OnceCell::new();
+    let store = TOKEN_FS_STORE.get_or_init(|| {
+        // TODO: Evaluate if this make sense, and what the possible attack vectors can be from relying on the
+        // untrusted fs and path functions.
```

Comment:
I think this initialization probably belongs outside of the enclave code: the untrusted app should be responsible for creating the directory, and the enclave should just receive the root path, assume it's valid, and let the `sgxfs` functions fail if anything goes wrong or unexpected.

Reply:
Easiest would probably be an (untrusted) environment variable?

We could also consider some kind of one-time enclave initialization / setup call, down the line, if we begin needing other kinds of startup configuration, but that's probably overkill for now.

(Given that we run in a relatively predictable Docker environment currently, we could probably even just hardcode the path at compile-time, for now, and defer using an environment variable until we actually need that.)

Reply:
Lets do that, since we need to do a pass for this regardless when we look at the other paths.